dali/imgcodec/decoders/libtiff/tiff_libtiff.cc

```diff
@@ -162,6 +166,12 @@ TiffInfo GetTiffInfo(TIFF *tiffptr) {
   LIBTIFF_CALL(TIFFGetFieldDefaulted(tiffptr, TIFFTAG_PLANARCONFIG, &planar_config));
   info.is_planar = (planar_config == PLANARCONFIG_SEPARATE);
 
+  if (info.is_palette) {
+    LIBTIFF_CALL(TIFFGetField(tiffptr, TIFFTAG_COLORMAP,
+                              &info.palette.red, &info.palette.green, &info.palette.blue));
+    info.channels = 3;  // Palette is always RGB
```

Comment:
Processing palette TIFFs is a bit tricky, because while they are in fact RGB images, their `SamplesPerPixel` is set to 1 (because, technically speaking, there's one value per pixel: the index). I set the number of input channels to 3 here to avoid confusion later.